src/services/employees.js

```diff
@@ -10,7 +10,9 @@ export const getEmployeesByType = async (type) => {
 };
 
 export const addEmployee = async (newEmployee) => {
-  const { data, error } = await api.post("/employees", newEmployee);
-  if (error) throw new Error("Employee could not be added");
+  const addPath = await api.post("/employees", newEmployee);
+  const editPath = await api.put("/employees", newEmployee);
+  const { data, error } = newEmployee.employeeId ? editPath : addPath;
+  if (error) throw new Error("Employee data could not be saved");
   return data;
```

Comment:
Should these two functionalities not be decoupled  ? I would like Delvin to decide .
But if we go ahead with the logic I think the newEmployee should be renamed to something generic for both add and edit, eg employeeData

Reply:
Yes, we can have separate functions for add, edit, delete, and so forth. Then when using tan stack query we can use those functions as needed. I generalize to name the functions after the http action that will be taken… add = post, update = put